src/Stack/Types/Package.hs

```diff
@@ -92,29 +92,103 @@ newtype ExeName = ExeName { unExeName :: Text }
-          ,packageTests :: !(Map Text TestSuiteInterface) -- ^ names and interfaces of test suites
-          ,packageBenchmarks :: !(Set Text)               -- ^ names of benchmarks
-          ,packageExes :: !(Set Text)                     -- ^ names of executables
-          ,packageOpts :: !GetPackageOpts                 -- ^ Args to pass to GHC.
-          ,packageHasExposedModules :: !Bool              -- ^ Does the package have exposed modules?
-          ,packageBuildType :: !BuildType                 -- ^ Package build-type.
+  Package {packageName :: !PackageName
+          -- ^ Name of the package.
+          -- This is PackageIdentifier pkgName in Cabal. 
+          ,packageVersion :: !Version
+          -- ^ Version of the package.
+          -- This is PackageIdentifier pkgVersion in cabal.
+          ,packageLicense :: !(Either SPDX.License License)
+          -- ^ The license the package was released under.
+          -- This is PackageDescription licenseRaw in cabal. 
+          ,packageFiles :: !GetPackageFiles
+          -- ^ Get all files of the package.
+          -- Absent in cabal.
+          ,packageDeps :: !(Map PackageName DepValue)
+          -- ^ Packages that the package depends on, both as libraries and build tools.
+          -- Derived from cabal:
+          --
+          --    - allBuildInfo (allBuildInfo') function (see main comment)
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:targetBuildDepends targetBuildDepends function>
+          --
+          -- see packageDependencies and packageDescTools for more details.
+          -- Note that this filters out _itself_ and all the non buildables. 
+          ,packageUnknownTools :: !(Set ExeName)
+          -- ^ Build tools specified in the legacy manner (build-tools:) that failed the hard-coded lookup.
+          -- Likely derived from cabal see the field just before.
+          ,packageAllDeps :: !(Set PackageName)
+          -- ^ Original dependencies (not sieved).
+          -- The keyset of package deps.
+          ,packageGhcOptions :: ![Text]
+          -- ^ Ghc options used on package.
+          -- Comes from 'PackageConfig' 'packageConfigGhcOptions'.
+          -- Likely derived from cabal in
+          --
+          --    - buildTypeRaw function (see main comment)
+          --    - <https://hackage.haskell.org/package/Cabal-3.2.1.0/docs/Distribution-PackageDescription.html#v:options options function>
+          --
+          -- see 'generateBuildInfoOpts', and then all the way from loadPackage.
+          ,packageCabalConfigOpts :: ![Text]
+          -- ^ Additional options passed to ./Setup.hs configure
+          -- Absent in cabal ? Comes from 'PackageConfig' 'packageConfigCabalConfigOpts'.
+          ,packageFlags :: !(Map FlagName Bool)
+          -- ^ Flags used on package.
+          -- Absent in cabal ?
```

Comment:
How could it be absent? it's just in a different place, see e.g. `flagAssignment` in https://downloads.haskell.org/ghc/8.10.3/docs/html/libraries/Cabal-3.2.1.0/Distribution-Simple-LocalBuildInfo.html#t:LocalBuildInfo